documentation/writing_tests.md

```diff
@@ -0,0 +1,1496 @@
+
+Some examples can be found at: 
+
+- [sdk/keyvault/test-resources.json](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/test-resources.json).
+- [sdk/storage/test-resources.json](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/test-resources.json).
+- [sdk/servicebus/test-resources.json](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/test-resources.json).
+
+The `tests.yml` file must be placed at the package folder. This file is in charge of specifying when to run the tests for this package, what environments to use to run the tests, and how the tests are executed. You can learn how to write these files by following the guide: [Creating live tests](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/48/Create-a-new-Live-Test-pipeline?anchor=creating-live-tests) [🔒][TIPS].
+
+Some examples can be found at: 
+
+- [sdk/keyvault/keyvault-keys/tests.yml](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-keys/tests.yml).
+- [sdk/storage/storage-blob/tests.yml](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/storage-blob/tests.yml).
+- [sdk/servicebus/service-bus/tests.yml](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/service-bus/tests.yml).
+
+These files will deal with the environment variables needed by your tests. Some of these environment variables are quite standard. Generally speaking, the SDK tests will use information from the tenant and the client of the resources that the tests are working with. To effectively provide these to the automated tests, we need to enable the pipeline to use some specific configuration.
+
+First you must go to https://dev.azure.com/azure-sdk/ [🔒][TIPS] and look for the builds that have been configured to target your project, then:
+
+- Click the pipeline you want to test on PR.
+- Press the "Edit" button at the top right corner.
+- Press the three dot menu at the top right corner. A menu will drop down. Press the "Triggers" option from that menu.
+- A page will load with an horizontal menu near the top-center with the following options: "YAML", "Variables", "Triggers" and "History".
+- Click the Variables option of that menu.
+- Click on "Variable Groups".
+- If the variable group you want to select is not visible in that page, add it: You'll see a button that will say "🔗 Link variable group". Click it, then use the "🔍 search" input and type the name `Secrets for Resource Provisioner`, and then link it.
+
+It should end up looking something like this:
+
+![image](https://user-images.githubusercontent.com/417016/72285413-f8e54700-363a-11ea-959e-cb1bc4c074ba.png)
```

Comment:
This is a screenshot of a Unified Pipeline, it should probably be a live tests pipeline.

Reply:
I'm going to link to [this doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=variable-groups) instead of having these steps here.